### DISCUSSION TITLE
Does Supabase support multi-region storage? (AWS S3 Multi region access points)

### DISCUSSION THREAD

**Discussion by www-link-inc** [supabase/supabase#29975] [Questions]
When I go to the Infrastructure settings page, it doesn't say anything about storage, just the database. AWS S3 does support [multi region access points](https://aws.amazon.com/s3/features/multi-region-access-points/), is that possible with Supabase?

If not, how would you suggest I handle storage in multiple regions? Should I just create a new project (project #2), and then when a user uploads an image, tell the database in project #1 where the image is located?

**Comment by GaryAustin1** [comment]
There is no option for the s3 storage not residing in the same region as your database.

I don't have specific suggestions on a way around it.  Your idea could work, but you would want to keep the JWT secrets in sync if you need RLS on your file operations.  Also when uploading you would have to decide when region to store in.   Storing in both regions could get complex if you have to deal with file updates/deletes.

**Reply by www-link-inc** [reply]
That's unfortunate. Are there any current plans to support multi region access points?

Ah yes, it would be tricky with JWT for RLS.

**Reply by GaryAustin1** [reply]
I can't comment on Supabase plans as not an employee, but in general they don't comment on roadmap type things except in feature announcements.
